File: Go/Day06.go
package main

import (
	"strings"
)

import Helper "./adventhelper"

func eachAnswer(ss string, split string) (n int, n2 int) {
	count := make(map[string]int)
	lines := strings.Split(ss, split)

	for _, l := range lines {
		for _, c := range l {
			count[string(c)] += 1
		}
	}

	n2 = 0
	for _, v := range count {
		if v == len(lines) {
			n2 += 1
		}
	}

	return len(count), n2
}

func main() {
	ss, _ := Helper.ParseLineGroups("../input/input06.txt", ":")

	part1 := 0
	part2 := 0
	for _, s := range ss {
		v1, v2 := eachAnswer(s, ":")
		part1 += v1
		part2 += v2
	}


	Helper.PrintSoln(6, part1, part2)

	return
}

File: Rust/src/day06.rs
use crate::advent_helper::advent_helper::parse_line_groups;

use std::collections::HashMap;

fn each_answer(s: &str, split_char: &str) -> (usize, usize)
{
	let mut count: HashMap<char, usize> = HashMap::new();
	let ws: Vec<&str> = s.split(split_char).collect();

	for w in &ws
	{
		for (_, c) in w.chars().enumerate()
		{
			*count.entry(c).or_insert(0) += 1;
		}
	}

	let mut n: usize = 0;
	for (_, value) in (&count).into_iter()
	{
    if *value == ws.len()
    {
    		n += 1;
    }
	}

	return (count.len(), n);
}

pub fn run()
{
	let v = parse_line_groups("../input/input06.txt", ":");

	let mut part1: usize = 0;
	let mut part2: usize = 0;
	for s in v
	{
		let (v1, v2) = each_answer(&s, ":");
		part1 += v1;
		part2 += v2;
	}

	println!("Day 6");
	println!("  Part 1: {}", part1);
	println!("  Part 2: {}", part2);

	return;
}
